Invalid plugin overrides panic on startup

### What happened?

Running `pulumi up` with an invalid `Pulumi.yaml["plugins"]["providers"]` causes a panic before main.

### Steps to reproduce

Run `pulumi up` with:

```yaml
name: csharp-artifactory
runtime: dotnet
description: A minimal C# Pulumi program
plugins:
  providers:
    - name: artifactory
      bin: ../../../bin # key should be 'path'
```

### Expected Behavior

A useful error message.

### Actual Behavior

```
X pulumi up
panic: fatal: could not find provider folder at path 

goroutine 1 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /Users/ianwahbe/go/src/github.com/pulumi/pulumi/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.AssertNoError({0x102720ae0?, 0xc0004c3990?})
        /Users/ianwahbe/go/src/github.com/pulumi/pulumi/sdk/go/common/util/contract/assert.go:40 +0x88
github.com/pulumi/pulumi/pkg/v3/codegen/schema.newPulumiPackage()
        /Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/schema/mock_pulumi_schema.go:41 +0x42d
github.com/pulumi/pulumi/pkg/v3/codegen/schema.init()
        /Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/schema/mock_pulumi_schema.go:45 +0x11f
```

### Output of `pulumi about`

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20230117--cli-display--fixes-11864-pulumi-panics-when-pulumi-yaml-provider-plugin-does-not-have-a-path-provided.yaml
+++ b/changelog/pending/20230117--cli-display--fixes-11864-pulumi-panics-when-pulumi-yaml-provider-plugin-does-not-have-a-path-provided.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: cli/display
+  description: "Fixes #11864. Pulumi panics before main when Pulumi.yaml provider plugin does not have a path provided."

--- a/sdk/go/common/workspace/project.json
+++ b/sdk/go/common/workspace/project.json
@@ -238,6 +238,11 @@
         "pluginOptions":{
             "title":"PluginOptions",
             "type":"object",
+            "additionalProperties":false,
+            "required":[
+                "name",
+                "path"
+            ],
             "properties":{
                 "name":{
                     "type":"string",

--- a/tests/integration/integration_test.go
+++ b/tests/integration/integration_test.go
@@ -644,3 +644,37 @@ func TestExcludeProtected(t *testing.T) {
 	stdout, _ = e.RunCommand("pulumi", "destroy", "--skip-preview", "--yes", "--exclude-protected")
 	assert.Contains(t, stdout, "There were no unprotected resources to destroy. There are still 7")
 }
+
+func TestInvalidPluginError(t *testing.T) {
+	t.Parallel()
+
+	e := ptesting.NewEnvironment(t)
+	defer func() {
+		if !t.Failed() {
+			e.DeleteEnvironment()
+		}
+	}()
+
+	pulumiProject := `
+name: invalid-plugin
+runtime: yaml
+description: A Pulumi program referencing an invalid plugin.
+plugins:
+  providers:
+    - name: fakeplugin
+      bin: ./does/not/exist/bin # key should be 'path'
+`
+
+	integration.CreatePulumiRepo(e, pulumiProject)
+	e.SetBackend(e.LocalURL())
+	{
+		_, stderr := e.RunCommandExpectError("pulumi", "stack", "init", "invalid-resources")
+		assert.NotContains(t, stderr, "panic: ")
+		assert.Contains(t, stderr, "error: ")
+	}
+	{
+		_, stderr := e.RunCommandExpectError("pulumi", "pre")
+		assert.NotContains(t, stderr, "panic: ")
+		assert.Contains(t, stderr, "error: ")
+	}
+}

